linter/ginkgo_linter.go

```diff
@@ -396,21 +401,25 @@ func doCheckExpression(pass *analysis.Pass, config types.Config, assertionExp *a
 
 	} else if checkMatchError(pass, assertionExp, actualArg, handler, reportBuilder, isExprError) {
 		return false
+	} else if checkSucceedSync(pass, assertionExp, actualArgs, handler, reportBuilder, bool(config.SuppressSucceed)) {
```

Comment:
If I understand what you're saying correctly, i.e. something like `if isSucceed(...) { return checkSucceed(...) }`, this will prevent the rules below from being checked even if the usage of Succeed is fine.

Also, most of the checks here work the same way of doing both structure check as well as reporting. What's special about Succeed?